framework/src/outputs/CSV.C

```diff
@@ -130,9 +151,28 @@ CSV::output(const ExecFlagType & type)
   // Output each VectorPostprocessor's data to a file
   if (_write_vector_table && processor_id() == 0)
   {
+
+    // The VPP table will not write the same data twice, so to get the symlinks correct
+    // for EXEC_FINAL (when other flags exist) whenever files are written the names must
+    // be stored. These stored names are then used outside of this loop when the EXEC_FINAL call is
+    // made.
+    _latest_vpp_filenames.clear();
+
     for (auto & it : _vector_postprocessor_tables)
     {
-      auto vpp_name = it.first;
+      const std::string & vpp_name = it.first;
+      std::string short_name = MooseUtils::shortName(vpp_name);
+
+      std::ostringstream output;
+      output << _file_base << "_" << short_name;
+
+      if (!vpp_data.containsCompleteHistory(vpp_name))
+        output << "_" << std::setw(_padding) << std::setprecision(0) << std::setfill('0')
```

Comment:
Why are you not calling the `getVectorPostprocessorFileName()` method?

Reply:
rebase likely undid this change